src/Layers/DynamicMapLayer.js

```diff
@@ -42,6 +42,12 @@ export var DynamicMapLayer = RasterLayer.extend({
   },
 
   setLayers: function (layers) {
+    if (!layers) { layers = []; }
+    if (layers.length === 0 && this.options.layers.length) {
```

Comment:
the second check is problematic when a dynamicMapLayer is instantiated _without_ a layers constructor parameter and `.setLayers([])` is called later.